pkg/monitortests/machines/watchmachines/machines.go

```diff
@@ -0,0 +1,68 @@
+package watchmachines
+
+import (
+	"context"
+	"time"
+
+	"github.com/openshift/origin/pkg/monitor/monitorapi"
+
+	machine "github.com/openshift/api/machine/v1beta1"
+	machineClient "github.com/openshift/client-go/machine/clientset/versioned"
+	machineinformerv1 "github.com/openshift/client-go/machine/informers/externalversions/machine/v1beta1"
+	"k8s.io/client-go/tools/cache"
+)
+
+func startMachineMonitoring(ctx context.Context, m monitorapi.RecorderWriter, client machineClient.Interface) {
+
+	machinePhaseChangeFns := []func(machine, oldMachine *machine.Machine) []monitorapi.Interval{
+		func(machine, oldMachine *machine.Machine) []monitorapi.Interval {
+			var intervals []monitorapi.Interval
+			now := time.Now()
+			if machine.Status.Phase != nil && oldMachine.Status.Phase != nil && *machine.Status.Phase != *oldMachine.Status.Phase {
+				intervals = append(intervals,
+					monitorapi.NewInterval(monitorapi.SourceMachine, monitorapi.Warning).
+						Locator(monitorapi.NewLocator().MachineFromName(machine.Name)).
+						Message(monitorapi.NewMessage().Reason(monitorapi.MachinePhaseChanged).
+							HumanMessage("changed")).
+						Build(now, now))
+			}
+			return intervals
+		},
+	}
+
+	machineInformer := machineinformerv1.NewMachineInformer(client, "openshift-machine-api", time.Hour, nil)
+	machineInformer.AddEventHandler(
+		cache.ResourceEventHandlerFuncs{
+			AddFunc: func(obj interface{}) {
+			},
+			DeleteFunc: func(obj interface{}) {
```

Comment:
I don't think the Delete actually transitions the phase, this is done on a status update later, so I expect this probably won't be too helpful

Reply:
I sorta assumed it is required to add these as boilerplate. I can drop it if its not needed.

Reply:
I added an empty function as I don't think it will be very useful.